Aurora Global Database Setup and other updates

catcharbind opened the pull request:
- Aurora Global database setup across 2 AWS regions
- Setup for both Aurora PostgreSQL and MySQL
- Parameter groups
- Enhanced Monitoring

tonynv commented on deploy/main.tf:
```diff
@@ -17,9 +17,13 @@ resource "random_string" "rand4" {
 module "vpc_label" {
-  source    = "aws-ia/label/aws"
-  version   = "0.0.2"
+  source    = "aws-quickstart/label/aws"
```
nothing should be referencing aws-quickstart (use aws-ia)

catcharbind replied:
Updated!

tonynv commented on deploy/main.tf:
```diff
@@ -32,25 +36,44 @@ module "vpc_label" {
-  source            = "aws-ia/vpc/aws"
-  version           = "0.0.2"
+module "aurora_vpc_p" {
+  source            = "aws-quickstart/vpc/aws"
```
nothing should be referencing aws-quickstart (use aws-ia)

tonynv commented on setup_workspace/workspace.tf:
```diff
@@ -10,8 +10,8 @@ resource "random_pet" "name" {
 module "tfc_workspace" {
-  source                = "aws-ia/cloud_workspace/hashicorp"
-  version               = "0.0.1"
+  source                = "aws-quickstart/tfc_workspace/aws"
```
nothing should be referencing aws-quickstart (use aws-ia)

tonynv commented on deploy/main.tf:
```diff
@@ -32,25 +36,44 @@ module "vpc_label" {
+}
+
+module "aurora_vpc_s" {
+  source            = "aws-quickstart/vpc/aws"
```
nothing should be referencing aws-quickstart (use aws-ia)

tonynv reviewed (changes requested):
Do not reference aws-quickstart, everything should be pointed to aws-ia

tonynv reviewed (commented):
By default it create the global db.  I suggest we create two entry points in the deploy folder deploy/standard_db and deploy/global_db  the current main should move into that folder. I have also updated the vpc module with create_vpc. With option we could potentialy keep a single entry point under deploy

tonynv commented:
Resource module.aurora.aws_kms_key.kms_p has lifecycle.prevent_destroy set. This prevent clean deletes

catcharbind commented:
Updated setup_globaldb and storage_encrypted to false by default. Updated VPC module version to create vpc conditionally.
KMS keys need to be retained for any encrypted Aurora snapshots to be retained.
